File: app/src/main/java/study/lowmans/mytest/APIService.java
package study.lowmans.mytest;

import java.util.List;

import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.GET;
import retrofit2.http.Path;

public class APIService {
    public static final String API_URL = "https://api.github.com";

    public static Retrofit retrofitIntance() {
       return   new Retrofit.Builder()
                .baseUrl(API_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

    }
    public static class Contributor {
        public final String login;
        public final int contributions;

        public Contributor(String login, int contributions) {
            this.login = login;
            this.contributions = contributions;
        }
    }

    public interface GitHub {
        @GET("/repos/{owner}/{repo}/contributors")
        Call<List<Contributor>> contributors(@Path("owner") String owner, @Path("repo") String repo);
    }
}

File: app/src/main/java/study/lowmans/mytest/APIService.kt
package study.lowmans.mytest

import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import retrofit2.Call

class APIService {
    interface GitHub {
        @GET("/repos/{owner}/{repo}/contributors")
        fun contributors(@Path("owner") owner: String, @Path("repo") repo: String): Call<List<APIData.Contributor>>
    }

    interface MyApi {
        @GET("/test.php")
        fun test (@Query("id") id: Int) :Call<APIData.Test>
    }
}